lib/ddtrace/span/external_resource.rb

```diff
@@ -0,0 +1,38 @@
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/http'
+
+module Datadog
+  class Span
+    # Extension for Datadog::Span that tracks if this span
+    # represents a unit of work performed by this application
+    # or an external resource being indirectly measured
+    # (e.g. SQL, Redis, outbound HTTP request).
+    module ExternalResource
+      attr_writer :external_resource
+
+      # Span types of known to represent an internal application trace
+      INTERNAL_APPLICATION_SPAN_TYPES = [
```

Comment:
I would suggest we maybe move this to the `Ext` namespace somehow, as it principally defines some constants we might want to generally refer/use without loading `ExternalResource`. It's also a little weird having `ExternalResource::INTERNAL_APPLICATION_SPAN_TYPES`, as these two elements seem contrasted to one another.

Perhaps it belongs in `Ext::AppTypes`, `Ext::Service::INTERNAL_SPAN_TYPES` or something else that captures its category well.